sklearn/cluster/tests/test_bicluster.py

```diff
@@ -208,26 +208,43 @@ def test_perfect_checkerboard():
 
 
 @pytest.mark.parametrize(
-    "args",
+    "args1",
     [
+        {"n_init": 0},
         {"n_clusters": (3, 3, 3)},
+        {"n_clusters": (6, 6)},
         {"n_clusters": "abc"},
         {"n_clusters": (3, "abc")},
+        {"n_clusters": ("abc", 3)},
         {"method": "unknown"},
         {"n_components": 0},
         {"n_best": 0},
         {"svd_method": "unknown"},
         {"n_components": 3, "n_best": 4},
     ],
 )
-def test_errors(args):
+def test_value_errors(args1):
```

Comment:
We don't need to test for the `ValueError` and `TypeError`. It can be part of the parametrization:

```python
@pytest.mark.parametrize(
    "params, type_err, err_msg",
    [
        ({"n_init": 0}, ValueError, "n_init == 0, must be >= 1."),
        ({"n_init": 1.5}, TypeError, "n_init must be an instance of"),
        (
            {"n_clusters": (3, 3, 3)},
            ValueError,
            r"Incorrect parameter n_clusters has value: \(3, 3, 3\)",
        ),
        (
            {"n_clusters": (6, 6)},
            ValueError,
            r"Incorrect parameter n_clusters has value: \(6, 6\)",
        ),
        (
            {"n_clusters": "abc"},
            ValueError,
            "Incorrect parameter n_clusters has value: abc",
        ),
        (
            {"n_clusters": (3, "abc")},
            ValueError,
            r"Incorrect parameter n_clusters has value: \(3, 'abc'\)",
        ),
        (
            {"n_clusters": ("abc", 3)},
            ValueError,
            r"Incorrect parameter n_clusters has value: \('abc', 3\)",
        ),
        ({"method": "unknown"}, ValueError, "Unknown method: 'unknown'"),
        ({"n_components": 0}, ValueError, "n_components == 0, must be >= 1."),
        ({"n_components": 1.5}, TypeError, "n_components must be an instance of"),
        ({"n_best": 0}, ValueError, "n_best == 0, must be >= 1."),
        ({"n_best": 1.5}, TypeError, "n_best must be an instance of"),
        ({"svd_method": "unknown"}, ValueError, "Unknown SVD method: 'unknown'"),
        ({"n_components": 3, "n_best": 4}, ValueError, "n_best == 4, must be <= 3."),
    ],
)
def test_spectalbiclustering_parameter_validation(params, type_err, err_msg):
    """Check parameters validation in `SpectralBiClustering`"""
    data = np.arange(25).reshape((5, 5))
    model = SpectralBiclustering(**params)
    with pytest.raises(type_err, match=err_msg):
        model.fit(data)
```

I don't see any test for `SpectralCoClustering`. We should have a similar test, however it should be limited to the available parameters.